iotedgehubdev/cli.py

```diff
@@ -189,28 +198,16 @@ def modulecred(modules, local, output_file):
               help='Docker daemon socket to connect to')
 @_with_telemetry
 def start(inputs, port, deployment, verbose, host):
-    configFile = HostPlatform.get_config_file_path()
-    try:
-        with open(configFile) as f:
-            jsonObj = json.load(f)
-            if CONN_STR in jsonObj and CERT_PATH in jsonObj and GATEWAY_HOST in jsonObj:
-                connection_str = jsonObj[CONN_STR]
-                cert_path = jsonObj[CERT_PATH]
-                gatewayhost = jsonObj[GATEWAY_HOST]
-                edgeManager = EdgeManager(connection_str, gatewayhost, cert_path)
-                if host is not None:
-                    os.environ[DOCKER_HOST] = host
-            else:
-                output.error('Missing keys in config file. Please run `iotedgehubdev setup` again.')
-                sys.exit(1)
-    except Exception as e:
-        raise e
+    edge_manager = _parse_config_json()
 
-    hostname_hash, suffix = Utils.hash_connection_str_hostname(connection_str)
-    telemetry.add_extra_props({'iothubhostname': hostname_hash, 'iothubhostnamesuffix': suffix})
+    if edge_manager:
+        if host is not None:
+            os.environ[DOCKER_HOST] = host
 
-    if inputs is None and deployment is not None:
-        try:
+        hostname_hash, suffix = Utils.hash_connection_str_hostname(edge_manager.hostname)
```

Comment:
better not access data member directly, Prefer to add an api in edgemanager like getHostName() as a declared protocol

Reply:
Added `hostname` property getter and made `hostname` private.